src/main/java/org/team1540/elmo/RobotContainer.java

```diff
@@ -58,29 +65,35 @@ public void setupMiscComponents() {
      */
     private void configureButtonBindings()
     {
-//        new JoystickButton(driver,XboxController.Button.kA.value).whenPressed(
-//                new EjectLower(innerEjector,outerEjector)
-//        );
+        Trigger eitherBumper = new JoystickButton(driver,XboxController.Button.kLeftBumper.value).or(new JoystickButton(driver,XboxController.Button.kRightBumper.value));
+        Trigger neitherBumper = eitherBumper.negate();
+
         // left side
-        new JoystickButton(driver,XboxController.Button.kX.value).whenPressed(
+        new JoystickButton(driver,XboxController.Button.kX.value).and(neitherBumper).whenActive(
                 new EjectInnerCommand(innerEjector)
         );
         // right side
-        new JoystickButton(driver,XboxController.Button.kA.value).whenPressed(
+        new JoystickButton(driver,XboxController.Button.kB.value).and(neitherBumper).whenActive(
                 new EjectOuterCommand(outerEjector)
         );
         // upper
-        new JoystickButton(driver,XboxController.Button.kY.value).whenPressed(
+        new JoystickButton(driver,XboxController.Button.kY.value).and(neitherBumper).whenActive(
                 new EjectUpperCommand(upperEjector)
         );
-        // retract all
-        new JoystickButton(driver,XboxController.Button.kB.value).whenPressed(
-                new ResetEjectorCommand(upperEjector).alongWith(
-                new ResetEjectorCommand(outerEjector).alongWith(
-                new ResetEjectorCommand(innerEjector)))
+
+        // toggle individual suction when either bumper and X:inner / B:outer / Y:upper are pressed
+        new JoystickButton(driver,XboxController.Button.kX.value).and(eitherBumper).toggleWhenActive(
+            new ResetEjectorAndWaitForeverCommand(innerEjector,true)
+        );
+        new JoystickButton(driver,XboxController.Button.kB.value).and(eitherBumper).toggleWhenActive(
+            new ResetEjectorAndWaitForeverCommand(outerEjector,true)
+        );
+        new JoystickButton(driver,XboxController.Button.kY.value).and(eitherBumper).toggleWhenActive(
+            new ResetEjectorAndWaitForeverCommand(upperEjector,true)
```

Comment:
This control setup is very clever, I assume it's just for testing though? Outside of testing, the pneumatics will only be controlled during auto

Reply:
Yeah testing D; Maybe there can be a button to extend the outer suction and pick up vertical standing tubes